awx/main/utils/common.py

```diff
@@ -1096,7 +1096,11 @@ def create_temporary_fifo(data):
     path = os.path.join(tempfile.mkdtemp(), next(tempfile._get_candidate_names()))
     os.mkfifo(path, stat.S_IRUSR | stat.S_IWUSR)
 
-    threading.Thread(target=lambda p, d: open(p, 'wb').write(d), args=(path, data)).start()
+    def tmp_write(path, data):
+        with open(path, 'wb') as f:
+            f.write(data)
+
+    threading.Thread(target=tmp_write, args=(path, data)).start()
```

Comment:
@AlanCoding another example of https://github.com/ansible/awx/pull/15620#discussion_r1837195000:
```suggestion
    threading.Thread(target=lambda p, d: Path(p).write_bytes(d), args=(path, data)).start()
```

This is one of the reasons I didn't really want to include fixing things into the scope of the PR, since a good solution might need to be iterated on, which may make the scope grow indefinitely.

Reply:
That's fine with me. I suspected `Path` could do this more clearly. It's your PR, you can include it or not.

Reply:
I didn't exactly expect you to merge it. I could have changed the base after your PR.

Reply:
@AlanCoding Ah, I was under the impression that you wanted it to be a part of the PR before merging. Anyway, it's not really blocking, I just wanted to be explicit about my observations. I think I'll add these code suggestions on top and then wait until somebody hits merge, then.

Reply:
Or maybe not… This import isn't there, and I won't be doing this from browser. Let's leave it as is and document somewhere that it's nice to address it.